ca_gazebo/src/color_plugin.cpp

```diff
@@ -0,0 +1,109 @@
+#include <string> //c++ libraries
+
+//ros libraries
+#include <ros/ros.h>
+#include <ros/console.h>
+#include <std_msgs/Bool.h>
+
+//gazebo libraries
+#include <gazebo/common/Plugin.hh>
+#include <gazebo/sensors/Sensor.hh>
+#include <gazebo/sensors/CameraSensor.hh>
+#include <gazebo/sensors/SensorTypes.hh>
+#include "gazebo_plugins/gazebo_ros_camera.h"
+
+//own libraries
+#include "ca_gazebo/color_sensor_plugin.h"
+
+namespace gazebo
+{
+// Register this plugin with the simulator
+GZ_REGISTER_SENSOR_PLUGIN(GazeboRosColor)
+
+////////////////////////////////////////////////////////////////////////////////
+// Constructor
+GazeboRosColor::GazeboRosColor() : nh_("color_sensor_plugin"),
+                                   _fov(6),
+                                   _range(10),
+                                   CameraPlugin(),
+                                   GazeboRosCameraUtils()
+
+{
+
+}
+
+GazeboRosColor::~GazeboRosColor()
+{
+  ROS_DEBUG_STREAM_NAMED("camera", "Unloaded");
+}
+
+bool GazeboRosColor::IsColor(const my_color &target, const my_color &color)
+{
+  return (abs(color[0] - target[0]) <= COLOR_TOLERANCE and abs(color[1] - target[1]) <= 
+  COLOR_TOLERANCE && abs(color[2] - target[2]) <= COLOR_TOLERANCE);
```

Comment:
Consider separating these into multiple lines so it is easier to read.

Reply:
Adressed